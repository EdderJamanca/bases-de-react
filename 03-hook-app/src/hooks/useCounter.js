import { useState } from "react"


export const useCounter = (initialState=10) => {

    const {state, setState}=useState(initialState); // por defecto recibe 10

    const incremento=(factor=1)=>{
        setState( state + factor )
    };

    const reset=()=>{
        setState(initialState)
    }

    const decremento =(factor=1)=>{
        setState( state - factor);
    };

    return {state, incremento,reset, decremento};
}
