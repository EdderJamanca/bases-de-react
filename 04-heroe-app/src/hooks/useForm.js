import { useState } from 'react'
 
export const useForm = (initialState={}) => {

   const [value, setvalue] = useState(initialState);

   const reset = ()=>{
       setvalue(
           initialState
       )
   }

   const handleImputChange =({target})=>{
       setvalue({
           ...value,
           [target.name ]:target.value
       })
   };

    return [value, handleImputChange, reset ];
}
