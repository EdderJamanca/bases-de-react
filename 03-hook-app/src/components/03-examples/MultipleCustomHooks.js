
import React from 'react';
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';
import './example.css';

const MultipleCustomHooks = () => {

    const {state, incremento} = useCounter(1);

    const {loading, data }= useFetch(`https://www.breakingbadapi.com/api/quotes/${ state }`);

    const { author, quote } =!!data && data[0];

    console.log(author, quote );
    

    return (
        <>
            <h1>BreakINGbAD Quotes</h1>
            <hr />
         
            {
                loading 
                ? (<div className="alert text-center alert-primary">
                   loading......
               </div>) : 
               (
                    <blockquote className="blockquote text-right">
                            <p className="mb-3">{quote}</p>
                            <footer className="blockquote-footer">{author}</footer>
                    </blockquote>
                    )

                    
            }
            <button onClick={()=>incremento()} className="btn btn-primary">
                otro
            </button>
        </>
    )
}

export default MultipleCustomHooks
