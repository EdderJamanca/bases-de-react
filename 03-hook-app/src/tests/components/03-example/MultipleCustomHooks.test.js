
import { shallow } from 'enzyme';
import React from 'react';
import MultipleCustomHooks from '../../../components/03-examples/MultipleCustomHooks';
import { useCounter } from '../../../hooks/useCounter';
import { useFetch } from '../../../hooks/useFetch';
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Pruebas en <MultipleCustomHoks/>',()=>{
    
    // beforeEach( ()=>{

    // })


    test('debe de mostrarse Correctamente', ()=>{
        
        useCounter.mockReturnValue({
            state: 10, 
            incremento:()=>{}
        })

        useFetch.mockReturnValue({
            data:null,
            loading:true,
            error:null
        })

        const wrapper =shallow( <MultipleCustomHooks />);
        expect(wrapper).toMatchSnapshot();

    });
    test('shauld debe de mostrar la informacion',()=>{
        useCounter.mockReturnValue({
            state: 10, 
            incremento:()=>{}
        })

        useFetch.mockReturnValue({
            data:[{
                author:'Fernando',
                quote:'Hola Mundo'
            }],
            loading:false,
            error:null
        });
        const wrapper= shallow(<MultipleCustomHooks />);
        expect( wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-3').text().trim()).toBe('Hola Mundo');
        expect( wrapper.find('.blockquote-footer').text().trim()).toBe('Fernando');

    })

})