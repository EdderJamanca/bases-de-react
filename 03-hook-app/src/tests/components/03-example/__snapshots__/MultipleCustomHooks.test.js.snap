// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHoks/> debe de mostrarse Correctamente 1`] = `
<Fragment>
  <h1>
    BreakINGbAD Quotes
  </h1>
  <hr />
  <div
    className="alert text-center alert-primary"
  >
    loading......
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    otro
  </button>
</Fragment>
`;
